patches/minecraft/net/minecraft/network/protocol/game/ClientboundAddMobPacket.java.patch

```diff
@@ -0,0 +1,44 @@
+--- a/net/minecraft/network/protocol/game/ClientboundAddMobPacket.java
++++ b/net/minecraft/network/protocol/game/ClientboundAddMobPacket.java
+@@ -21,6 +_,7 @@
+    private final byte f_131540_;
+    private final byte f_131541_;
+    private final byte f_131542_;
++   private final int[] partIds;
+ 
+    public ClientboundAddMobPacket(LivingEntity p_131545_) {
+       this.f_131531_ = p_131545_.m_142049_();
+@@ -40,6 +_,7 @@
+       this.f_131537_ = (int)(d1 * 8000.0D);
+       this.f_131538_ = (int)(d2 * 8000.0D);
+       this.f_131539_ = (int)(d3 * 8000.0D);
++      this.partIds = p_131545_.isMultipartEntity() ? java.util.Arrays.stream(p_131545_.getParts()).mapToInt(net.minecraft.world.entity.Entity::m_142049_).toArray() : new int[0];
+    }
+ 
+    public ClientboundAddMobPacket(FriendlyByteBuf p_178566_) {
+@@ -55,6 +_,7 @@
+       this.f_131537_ = p_178566_.readShort();
+       this.f_131538_ = p_178566_.readShort();
+       this.f_131539_ = p_178566_.readShort();
++      this.partIds = p_178566_.m_130100_();
+    }
+ 
+    public void m_5779_(FriendlyByteBuf p_131554_) {
+@@ -70,6 +_,7 @@
+       p_131554_.writeShort(this.f_131537_);
+       p_131554_.writeShort(this.f_131538_);
+       p_131554_.writeShort(this.f_131539_);
++      p_131554_.m_130089_(this.partIds);
```

Comment:
This is not suitable for FS -> VC. This will most definitely throw an `IndexOutOfBoundsException` when loading a multipart entity. You will need to implement a network filter to handle connections to a vanilla client with this method. 

Otherwise, a different solution would need to be used such as deferring the creation of the entity id until when the entity is added to a world.

Reply:
I think you mean VanillaServer and ForgeClient (With a ForgeServer and a VanillaClient the client will just ignore the extra data that is being sent).
I would add some `if (Minecraft.getInstance().getCurrentServer().forgeData.type.equals("VANILLA"))` statements to check if the server is a vanilla server . I would create a helper method for that in ForgeHooksClient to handle the sidedness of `Minecraft`. Or did you mean something else with a network filter? Or is there already a helper method to check if the server is a vanilla server?

Reply:
You also have to handle forge servers that do not have this patch yet.
And please verify that additional data is actually ignored - I am not certain on that.

Reply:
I checked the code and as I suspected: extra data is _not_ ignored:
> `throw new IOException("Packet " + p_130535_.channel().attr(Connection.ATTRIBUTE_PROTOCOL).get().getId() + "/" + j + " (" + packet.getClass().getSimpleName() + ") was larger than I expected, found " + friendlybytebuf.readableBytes() + " bytes extra whilst reading packet " + j);`

Reply:
Hm ok. That means I have to check if the connected client is a forge client and what forge version the client is. But I would have to do this inside the `ClientboundAddMobPacket.write` method where I don't know to which client the packet will be sent to. So do you have an idea how I could handle the server side of this problem.

Reply:
I found the `ForgeConnectionNetworkFilter` class. So this is what you meant with network filter. Will work on a fix for this problem now.

Reply:
Changed the PR to not make changes to the Vanilla network protocol.